linked_list/linked_list.py

```diff
@@ -10,70 +10,145 @@ def __init__(self, value, next_node = None):
 class LinkedList:
     def __init__(self):
       self.head = None # keep the head private. Not accessible outside this class
-
+    
     # returns the value in the first node
     # returns None if the list is empty
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
     def get_first(self):
-        pass
+        if self.head == None:
+            return None
+        return self.head.value
+        
+
 
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(1)
+    # Space Complexity: O(1)
     def add_first(self, value):
-        pass
+        new_node = Node(value)
+        new_node.next = self.head
+        self.head = new_node
+        
 
     # method to find if the linked list contains a node with specified value
     # returns true if found, false otherwise
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def search(self, value):
-        pass
+        current = self.head
+        if current == None:
+            return False
+        while current:
+            if current.value == value:
+                return True
+            else:
+                current = current.next
+        return False
 
     # method that returns the length of the singly linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def length(self):
-        pass
+        counter = 0
+        current = self.head
+        while current:
+            counter += 1
+            current = current.next
+        return counter
+
 
     # method that returns the value at a given index in the linked list
     # index count starts at 0
     # returns None if there are fewer nodes in the linked list than the index value
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def get_at_index(self, index):
-        pass
+        current = self.head
+        if current == None:
+            return None
+        counter = 0
+        while counter != index:
+            current = current.next
+            counter += 1

```

Comment:
You should also have a test in here to check to see if current is None.  If it is you can return None.

It's possible that `index` is > the length of the list.  Something the tests didn't catch.